x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyoutBody.tsx

```diff
@@ -0,0 +1,270 @@
+              defaultMessage: 'Choose the service you want to configure.'
+            }
+          )}
+        >
+          <EuiSelect
+            isLoading={serviceNamesStatus === 'loading'}
+            hasNoInitialSelection
+            options={serviceNames.map(text => ({ text }))}
+            value={serviceName}
+            disabled={Boolean(selectedConfig)}
+            onChange={e => {
+              e.preventDefault();
+              setServiceName(e.target.value);
+              setEnvironment(undefined);
+            }}
+          />
+        </EuiFormRow>
+
+        <EuiFormRow
+          label={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.serviceEnvironmentSelectLabel',
+            {
+              defaultMessage: 'Environment'
+            }
+          )}
+          helpText={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.serviceEnvironmentSelectHelpText',
+            {
+              defaultMessage:
+                'Only a single environment per configuration is supported.'
+            }
+          )}
+          error={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.serviceEnvironmentSelectErrorText',
+            {
+              defaultMessage:
+                'Must select a valid environement to save a configuration.'
+            }
+          )}
+          isInvalid={
+            !(
+              selectedConfig ||
+              (!selectedConfig &&
+                environment &&
+                isSelectedEnvironmentValid &&
+                environmentStatus === 'success') ||
+              isNaN(sampleRate)
+            )
+          }
+        >
+          <EuiSelect
+            key={serviceName} // rerender when serviceName changes to mitigate initial selection bug in EuiSelect
+            isLoading={environmentStatus === 'loading'}
+            hasNoInitialSelection
+            options={environmentOptions}
+            value={environment}
+            disabled={!serviceName || Boolean(selectedConfig)}
+            onChange={e => {
+              e.preventDefault();
+              setEnvironment(e.target.value);
+            }}
+          />
+        </EuiFormRow>
+
+        <EuiTitle size="xs">
+          <h3>
+            {i18n.translate(
+              'xpack.apm.settings.cm.flyOut.configurationSectionTitle',
+              {
+                defaultMessage: 'Configuration'
+              }
+            )}
+          </h3>
+        </EuiTitle>
+
+        <EuiSpacer size="m" />
+
+        <EuiFormRow
+          label={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.sampleRateConfigurationInputLabel',
+            {
+              defaultMessage: 'Transaction sample rate'
+            }
+          )}
+          helpText={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.sampleRateConfigurationInputHelpText',
+            {
+              defaultMessage:
+                'Choose a rate between 0.00 and 1.0. Default configuration is 1.0 (100% of traces).'
```

Comment:
Should say `0.000` since we added another decimal to the validation for 0.1% sample rate.